src/extensions/step_handlers/script_handler/inc/aduc/script_handler.hpp

```diff
@@ -55,4 +69,15 @@ class ScriptHandlerImpl : public ContentHandler
     static ADUC_Result PerformAction(const std::string& action, const tagADUC_WorkflowData* workflowData);
 };
 
+/**
+ * @brief Perform the specified action.
+ *
+ * @param action Name of the action to perform
+ * @param workflowData The workflow data
+ * @param prepareArgsOnly if true, only prepare the arguments and return them in the result
+ * @return ADUC_PerformAction_Results The result of the action and outputs of the script.
+ */
+ADUC_PerformAction_Results
+ScriptHandler_PerformAction(const std::string& action, const tagADUC_WorkflowData* workflowData, bool prepareArgsOnly);
```

Comment:
not thrilled with passing action as a string -- wouldn't this be better as an enum value?